### DISCUSSION TITLE
Using wrapper in an application

### DISCUSSION THREAD

**Discussion by jpdosher** [microsoft/powerplatform-cli-wrapper#446] [Q&A]
I posted this into Power Apps community forum, but got no responses... Is this possible with CLI Wrapper?

"Im building a tool to refactor a Power Apps canvas app (refactor controls names).
For now, it is a simple Python script the has simple functionality, but works. 👌
Now, I need to make it useful for citizen developers. I have a bunch of ideas, one of those is to create a GUI (user interface) to allow user to update a YourApp.msapp file, the system unpack this app, perform the refactoring, and unpack it again. Here is the deal, how I do that?!  How can i use power platform CLI inside a system package to run some CLI commands?"

**Comment by skfd** [comment]
Your refactoring script should only operate on the unpacked code of the Canvas App. User can handle unpacking and publishing the solutions themselves. (You can provide example scripts)

If you want you tool to say CLI -- Write it in powershell/c# and packed it as a [.net tool](https://learn.microsoft.com/en-us/dotnet/core/tools/global-tools). This way we can easily install and run it.

Alternatively make a UI tool for [XrmToolBox](https://www.xrmtoolbox.com/)?
